multicluster/controllers/multicluster/member_clusterset_controller.go

```diff
@@ -92,6 +92,9 @@ func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 		klog.InfoS("Received ClusterSet delete", "clusterset", req.NamespacedName)
 		if r.remoteCommonArea != nil {
+			if err := r.deleteMemberClusterAnnounce(ctx); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to delete MemberClusterAnnounce in the leader cluster: %v", err)
```

Comment:
Could we add a comment to say: "MemberClusterAnnounce could be kept in the leader cluster, if mc-controller in the member crashes after the failure".

@luolanzone : let me know if you think this can be fixed. But I feel we should put a comment before we really fix it.

Reply:
@hjiajing  is also working on a stale MemberClusterAnnounce clean up, I thin we can add comment to clarify that stale resources will be cleaned up by stale controller.

Reply:
I feel this cannot be solved by stale resources clean up. Leader cluster does not know MemberClusterAnnounce should be removed. Member cluster does not know it should connect to leader and remove the MemberClusterAnnounce if ClusterSet/Claims were already deleted in member. @luolanzone

Reply:
When ClusterSet is deleted from member, member controller will stop updating the timestamp annotation of MemberClusterAnnounce, I think the stale controller in the leader cluster can check if timestamp is no longer updated over 1 mins(or 5mins ?), then we can delete it.

Reply:
Ok. That sounds good to me. We can use less aggressive strategy to wait longer before removing the Announce. Anyway, could we add a comment here, says ""MemberClusterAnnounce could be kept in the leader cluster, if mc-controller in the member crashes after the failure. Leader cluster will have a garbage collection mechanism to delete the stale MemberClusterAnnounce.".

Reply:
@luolanzone My stale controller (not pushed) does not check the timestamp. It's good to check by timestamp. I will update my code and push later.

Reply:
@jianjuns Sure. The PR is #4054. I will add the comment after rebase. Thanks.

Reply:
A reminder for adding the suggested comment.